tools/Google.Cloud.Tools.Analyzers/InternalOptionalParametersRequiredCodeFixProvider.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Google.Cloud.Tools.Analyzers
+{
+    /// <summary>
+    /// Code fix provider for the <see cref="InternalOptionalParametersRequiredAnalyzer"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(InternalOptionalParametersRequiredCodeFixProvider)), Shared]
+    public class InternalOptionalParametersRequiredCodeFixProvider : CodeFixProvider
+    {
+        private const string Title = "Insert optional arguments";
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get { return ImmutableArray.Create(InternalOptionalParametersRequiredAnalyzer.DiagnosticId); }
+        }
+
+        public sealed override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var semanticModel = await document.GetSemanticModelAsync(context.CancellationToken).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var argumentListExpression =
+                root.FindToken(context.Span.Start).Parent.AncestorsAndSelf().OfType<ArgumentListSyntax>().FirstOrDefault();
+            if (argumentListExpression == null)
+            {
+                return;
+            }
+
+            var invocation =
+                semanticModel.GetOperation(argumentListExpression, context.CancellationToken) as IInvocationExpression;
+            if (invocation == null)
+            {
+                return;
+            }
+
+            var variablesInScope =
+                InternalOptionalParametersRequiredAnalyzer.GetVariablesInScope(argumentListExpression, semanticModel);
+
+            var omittedParameterVariablePairs = new List<Tuple<IParameterSymbol, ISymbol>>();
```

Comment:
It looks like we only actually use the names... can we use a `Tuple<string, string>` instead?